onnxruntime/contrib_ops/cpu/transformers/beam_search_impl_base.h

```diff
@@ -23,17 +23,19 @@ struct BeamSearchState : IBeamSearchState<T> {
     size_t next_token_size = SafeInt<size_t>(batch_beam_size) * parameters.vocab_size;
     this->next_token_logits = AllocateBuffer<T>(allocator, next_token_logits_buffer_, next_token_size);
     this->next_token_scores = AllocateBuffer<float>(allocator, next_token_scores_buffer_, next_token_size);
-
     this->next_tokens = AllocateBuffer<int32_t>(allocator, next_tokens_buffer_, SafeInt<size_t>(2) * batch_beam_size);
-
     this->next_indices = AllocateBuffer<int32_t>(allocator, next_indices_buffer_, SafeInt<size_t>(2) * batch_beam_size);
-
     this->next_scores = AllocateBuffer<float>(allocator, next_scores_buffer_, SafeInt<size_t>(2) * batch_beam_size);
 
     constexpr size_t max_parts_of_vocab = 128;
     size_t topk_buffer_size = SafeInt<size_t>(batch_beam_size) * (max_parts_of_vocab + 1) * parameters.num_beams * 2 * 2;
     this->topk_buffer = AllocateBuffer<float>(allocator, topk_temp_buffer_, topk_buffer_size);
 
+    if (allocator->Info().device.Type() == OrtDevice::GPU) {
+      size_t sequences_bytes = SafeInt<size_t>(2) * batch_beam_size * parameters.max_length;
```

Comment:
This is not bytes. Rename to sequences_elements?

Reply:
Good point, I was thinking sequences_size (to match the others) but can use sequences_elements